src/coreclr/src/System.Private.CoreLib/src/System/Collections/Generic/ComparerHelpers.cs

```diff
@@ -120,28 +120,33 @@ internal static object CreateDefaultEqualityComparer(Type type)
             object? result = null;
             var runtimeType = (RuntimeType)type;
 
-            // Specialize for byte so Array.IndexOf is faster.
             if (type == typeof(byte))
             {
+                // Specialize for byte so Array.IndexOf is faster.
                 result = new ByteEqualityComparer();
             }
-            // If T implements IEquatable<T> return a GenericEqualityComparer<T>
+            else if (type == typeof(string))
+            {
+                // Specialize for string, as EqualityComparer<string>.Default is on the startup path
+                result = new GenericEqualityComparer<string>();
+            }
             else if (type.IsAssignableTo(typeof(IEquatable<>).MakeGenericType(type)))
             {
+                // If T implements IEquatable<T> return a GenericEqualityComparer<T>
                 result = CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(GenericEqualityComparer<int>), runtimeType);
```

Comment:
```suggestion
                result = CreateInstanceForAnotherGenericParameter((RuntimeType)typeof(GenericEqualityComparer<string>), runtimeType);
```
The template can be any instantiation. Better to reuse instance that we are using for other reasons.

Reply:
Done. Thanks.